receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processscraper
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/host"
+	"github.com/shirou/gopsutil/process"
+	"go.opencensus.io/trace"
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/processor/filterset"
+)
+
+// scraper for Process Metrics
+type scraper struct {
+	config    *Config
+	startTime pdata.TimestampUnixNano
+	includeFS filterset.FilterSet
+	excludeFS filterset.FilterSet
+
+	getProcessHandles func() (processHandles, error)
+}
+
+type processMetadata struct {
+	pid      int32
+	name     string
+	handle   processHandle
+	username string
+	cmdline  string
+}
+
+type processTimes struct {
+	*processMetadata
+	times *cpu.TimesStat
+}
+
+type processMemoryInfo struct {
+	*processMetadata
+	memoryInfo *process.MemoryInfoStat
+}
+
+type processIoCounters struct {
+	*processMetadata
+	ioCounters *process.IOCountersStat
```

Comment:
These structs are relatively small, would maybe embed processMetadata as a value instead and see if it improves benchmarks any.

Reply:
I've committed this change, though it's too hard to tell if made any impact to the benchmarks (at least on Windows) as this change is negligible compared to work done in the syscalls (which are not very consistent).